p2p/client.go

```diff
@@ -82,15 +84,17 @@ type Client struct {
 //
 // Basic checks on parameters are done, and default parameters are provided for unset-configuration
 // TODO(tzdybal): consider passing entire config, not just P2P config, to reduce number of arguments
-func NewClient(conf config.P2PConfig, privKey crypto.PrivKey, chainID string, logger log.Logger) (*Client, error) {
+func NewClient(conf config.P2PConfig, privKey crypto.PrivKey, chainID string, gater *conngater.BasicConnectionGater, logger log.Logger) (*Client, error) {
```

Comment:
Do we really need to pass it here? It's concrete type anyways, we probably should just create it in constructor, and then code can be simplified (we won't need to check if `gater` is `nil` everywhere).

Reply:
That was what I did at first, but I figured you might want it optional. So what you're saying is you don't need it to be optional?

Reply:
There are two things here: 1) optionality., 2) passing gater to Client vs creating Gater in client.

After giving this a second thought - lets leave this as is - it's better to pass Gater to client. Because in the `Node` we will have access to actual datasource, and it makes more sense to pass Connection Gater to P2P client, than to pass entire datasource.

I'm marking this conversation as resolved - no changes required :+1: